DataStorage.py

```diff
@@ -0,0 +1,86 @@
+from typing import Dict, List
+import operator
+import copy
+
+def remove_from_list(container, value):
+    try:
+        container.remove(value)
+    except ValueError:
+        pass
+    return container
+
+
+def pop_from_container(container, value):
+    try:
+        container.pop(value)
+    except ValueError:
+        pass
+    return container
+
+
+def update_dict(dictionary, entries):
+    dictionary.update(entries)
+    return dictionary
+
+# functions callable on storable data on the server by clients
+modify_functions = {
+    "add": operator.add,  # add together two objects, using python's "+" operator (works on strings and lists as append)
+    "mul": operator.mul,
+    "mod": operator.mod,
+    "max": max,
+    "min": min,
+    "replace": lambda old, new: new,
+    "default": lambda old, new: old,
+    "pow": operator.pow,
+    # bitwise:
+    "xor": operator.xor,
+    "or": operator.or_,
+    "and": operator.and_,
+    "left_shift": operator.lshift,
+    "right_shift": operator.rshift,
+    # lists/dicts
+    "remove": remove_from_list,
+    "pop": pop_from_container,
+    "update": update_dict,
+}
+
+class DataStorage:
+    stored_data: Dict[str, object]
+
+    def __init__(self, stored_data: Dict[str, object]):
+        self.stored_data = stored_data
+
+    def is_valid_set_cmd(self, set_cmd: Dict[str, object]) -> bool:

```

Comment:
```suggestion
    @static_method
    def is_valid_set_cmd(set_cmd: Dict[str, object]) -> bool:
```

But I think the proper way would be to just run the `set`, and raise and catch specific exceptions to return specific errors from multiserver.

Reply:
Hmm in C# managing by exceptions is always considered a bad pattern, but maybe this is a more proper thing todo in python

Reply:
yeah i noticed the error, ill rewrite it using an exception as black sliver suggested

Reply:
> `@static_method` doesn't exist. `@staticmethod` does.

Whoops

Re error handling in python: it's free if it doesn't see an error (since 3.11), so if non-error is the expected code path, exceptions are good in python. Catch also got cheaper in 3.11. Last but not least, accessing a collection uses raise already, so by using try-except you can often skip duplicate checks of error conditions.

Reply:
Oke well i did rewrite it using exceptions